src/formatting/comment.rs

```diff
@@ -1422,8 +1422,10 @@ impl<'a> Iterator for LineClasses<'a> {
             None => unreachable!(),
         };
 
+        let mut prev_kind = FullCodeCharKind::Normal;
         while let Some((kind, c)) = self.base.next() {
             // needed to set the kind of the ending character on the last line
+            prev_kind = self.kind;
```

Comment:
Could you swap these, just to maintain the association of the inline comment with the same line?

 ```suggestion
             prev_kind = self.kind;
            // needed to set the kind of the ending character on the last line
```